qa/tasks/cephadm.py

```diff
@@ -100,6 +100,43 @@ def update_archive_setting(ctx, key, value):
         yaml.safe_dump(info_yaml, info_file, default_flow_style=False)
 
 
+@contextlib.contextmanager
+def nvmeof_gateway_cfg(ctx, config):
+    source_host = config.get('source')
+    target_host = config.get('target')
+    nvmeof_service = config.get('service')
+    assert source_host
+    assert target_host
+    assert nvmeof_service
+    (remote,) = ctx.cluster.only(source_host).remotes.keys()
+    ip_address = remote.ip_address
+    r = remote.run(args=[
+        'systemctl', 'list-units',
+        run.Raw('|'), 'grep', nvmeof_service,
+        run.Raw('|'), 'awk', '{print $1}',
+        run.Raw('|'),
+        'sed', '-e', 's/[^@]*@//', '-e', 's/\.service$//',
```

Comment:
It might be slightly more friendly in terms of failure analysis to stop after the `grep` here, and then process the resulting string in plain python